config/volume.go

```diff
@@ -208,12 +253,48 @@ func (c *Client) ListAllVolumes() ([]string, error) {
 	return ret, nil
 }
 
-// WatchVolumes watches the volumes tree and returns data back to the activity channel.
-func (c *Client) WatchVolumes(activity chan *watch.Watch) {
+// WatchVolumeRuntimes watches the runtime portions of the volume and yields
+// back any information received through the activity channel.
+func (c *Client) WatchVolumeRuntimes(activity chan *watch.Watch) {
 	w := watch.NewWatcher(activity, c.prefixed(rootVolume), func(resp *client.Response, w *watch.Watcher) {
 		vw := &watch.Watch{Key: strings.Replace(resp.Node.Key, c.prefixed(rootVolume)+"/", "", -1), Config: nil}
 
-		if !resp.Node.Dir {
+		if !resp.Node.Dir && path.Base(resp.Node.Key) == "runtime" {
+			log.Debugf("Handling watch event %q for volume %q", resp.Action, vw.Key)
+			if resp.Action != "delete" {
+				volume := &RuntimeOptions{}
+
+				if resp.Node.Value != "" {
+					if err := json.Unmarshal([]byte(resp.Node.Value), volume); err != nil {
+						log.Errorf("Error decoding volume %q, not updating", resp.Node.Key)
+						time.Sleep(1 * time.Second)
+						return
+					}
+
+					if err := volume.Validate(); err != nil {
+						log.Errorf("Error validating volume %q, not updating", resp.Node.Key)
+						time.Sleep(1 * time.Second)
+						return
+					}
+					policy, vol := path.Split(path.Dir(resp.Node.Key))
+					vw.Key = path.Join(path.Base(policy), vol)
+					vw.Config = volume
+				}
+			}
+
+			w.Channel <- vw
+		}
+	})
+
+	watch.Create(w)
+}
+
+// WatchVolumeCreates watches the volumes tree and returns data back to the activity channel.
+func (c *Client) WatchVolumeCreates(activity chan *watch.Watch) {
```

Comment:
This is unused code I think, but it was just intended to provide completeness to the interface. Perhaps I should remove it?